src/adaptor/ini.rs

```diff
@@ -0,0 +1,502 @@
+
+use nom::{IResult, space, alphanumeric, multispace};
+
+use super::{Adaptor, Config, AdaptorError};
+
+use std::collections::{HashMap, hash_map};
+use std::fmt::Debug;
+use std::io::{Read, Write};
+use std::iter::FromIterator;
+use std::str;
+
+/// The adaptor struct for INI files
+/// Later, this might contain parameters for the myriad INI quirks
+pub struct IniAdaptor {
+    
+}
+
+impl IniAdaptor {
+    /// Constructs a new `IniAdaptor`
+    pub fn new() -> IniAdaptor {
+        IniAdaptor { }
+    }
+}
+
+impl<'a> Adaptor<'a> for IniAdaptor {
+    /// Deserialize the INI data into the `Config` AST
+    fn deserialize<R>(&self, mut reader: R) -> Result<Config, AdaptorError> 
+        where R: Read 
+    {
+        let mut buffer = Vec::new();
+        reader.read_to_end(&mut buffer);
+
+        // parse the basic INI structure
+        let output = match sections(&buffer) {
+            IResult::Done(_, o) => o,
+            _ => return Err("unable to parse INI data"),
+        };
+
+        let mut combined = HashMap::new();
+
+        // Here we convert the INI into our configuration AST,
+        // performing section and key de-duplication as necessary
+        for (name, pairs) in output {
+            // fetch existing entry or create a new one, deduplicating sections
+            let mut entry = combined.entry(name.into()).or_insert_with(|| HashMap::new());
+            // later, we will need schema data in order to encode type information into the AST
```

Comment:
Yes, we can use heuristics, although it could lead to possible situations where we incorrectly identify the type. That could be rectified with type coercion.